agent/api/container/status/managedagentstatus.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package status
+
+import (
+	"errors"
+	"strings"
+)
+
+const (
+	// ManagedAgentStatusNone is the zero state of a managed agent
+	ManagedAgentStatusNone ManagedAgentStatus = iota
+	// ManagedAgentCreated represents a managed agent which has dependencies in place
+	ManagedAgentCreated
+	// ManagedAgentRunning represents a managed agent that has started sucessfully
+	ManagedAgentRunning
+	// ManagedAgentStopped represents a managed agent that has stopped
+	ManagedAgentStopped
+)
+
+// ManagedAgentStatus is an enumeration of valid states in the managed agent lifecycle
+type ManagedAgentStatus int32
+
+var managedAgentStatusMap = map[string]ManagedAgentStatus{
+	"NONE":    ManagedAgentStatusNone,
+	"CREATED": ManagedAgentCreated,
+	"RUNNING": ManagedAgentRunning,
+	"STOPPED": ManagedAgentStopped,
+}
+
+// String returns a human readable string representation of this object
+func (mas ManagedAgentStatus) String() string {
+	for k, v := range managedAgentStatusMap {
+		if v == mas {
+			return k
+		}
+	}
+	return "NONE"
+}
+
+// Terminal returns true if the managed agent status is STOPPED
+func (mas ManagedAgentStatus) Terminal() bool {
+	return mas == ManagedAgentStopped
+}
+
+// IsRunning returns true if the managed agent status is either RUNNING
+func (mas ManagedAgentStatus) IsRunning() bool {
+	return mas == ManagedAgentRunning
+}
+
+// BackendStatus maps the internal managedAgent status in the agent to that in the backend
+func (mas ManagedAgentStatus) BackendStatus() string {
```

Comment:
What's the purpose of this? I don't think there's a possible way for this method to return `"PENDING"`. This is because it will always return `mas.String()`, so in effect this is a weird alias for `mas.String()`

https://goplay.corp.amazon.com/p/0766add6194da9cf06b11cee6ad9a7b7ef1cd411

Reply:
ManagedAgentCreated will return "PENDING" (because explicit `fallthrough`)

Reply:
`fallthrough` means kind of "execute the case below instead, which is 
```golang
ManagedAgentRunning:
    return mas.String()
```
Demo: https://play.golang.org/p/2Dk8zY4WgQj

Reply:
I made a change to this and to the unit test:
Backend status should always return a valid backend status.  This is just meant to force-map the status to one of `PENDING|RUNNING|STOPPED`